core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1927,6 +1928,10 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
     for {
       googleProjectId <- traceWithParent("getGoogleProjectFromBuffer", span)(_ => resourceBufferService.getGoogleProjectFromBuffer(projectPoolType, workspaceId))
       _ <- traceWithParent("updateGoogleProjectBillingAccount", span)(_ => gcsDAO.updateGoogleProjectBillingAccount(googleProjectId, Option(billingAccount)))
+      _ <- traceWithParent("updateGoogleProjectDisplayName", span)(_ => gcsDAO.updateGoogleProjectName(googleProjectId, workspaceName.toString))
+      _ <- traceWithParent("addGoogleProjectLabel: namespace", span)(_ => gcsDAO.addGoogleProjectLabel(googleProjectId, "workspaceNamespace", workspaceName.namespace))
+      _ <- traceWithParent("addGoogleProjectLabel: name", span)(_ => gcsDAO.addGoogleProjectLabel(googleProjectId, "workspaceName", workspaceName.name))
+      _ <- traceWithParent("addGoogleProjectLabel: id", span)(_ => gcsDAO.addGoogleProjectLabel(googleProjectId, "workspaceId", workspaceId))
```

Comment:
As noted above, I think we can consolidate all the label calls into a single call.  Wondering if we can't also consolidate all of the calls to update the Google Project into a single call.  So set the Billing Account, Name, Labels all into a single `updateGoogleProject` call.  

Taking it a step further, `addGoogleProjectLabel` calls `getGoogleProject` as does `getGoogleProjectNumber`.  Could we optimize to make one call to get the Google Project (and therefore the Google Project Number) and then we can use that instance to make any updates to set the Labels, Billing Account, and Name, and then make one single call back to Google to update the Google Project.  

I think reducing the amount of chatter back and forth with Google is something we should generally aim for whenever we can to reduce overall latency, potential quota issues, and just overhead in general.

Reply:
Yeah, I had been thinking about that. I think it is also safer from a management perspective (what happens if an asynchronous call futzes with the labels.) Was thinking it would be cleanest just to have an `updateGoogleProject` call, but for some reason that wasn't the current pattern in the codebase, so I tried to stick with what was existing. Much happier combining though.